mypy/stubgen.py

```diff
@@ -1071,6 +1083,48 @@ def process_namedtuple(self, lvalue: NameExpr, rvalue: CallExpr) -> None:
                 self.add(f"{self._indent}    {item}: Incomplete\n")
         self._state = CLASS
 
+    def is_typeddict(self, expr: Expression) -> bool:
+        if not isinstance(expr, CallExpr):
+            return False
+        callee = expr.callee
+        return (
+            isinstance(callee, NameExpr) and self.refers_to_fullname(callee.name, TPDICT_NAMES)
+        ) or (
+            isinstance(callee, MemberExpr)
+            and isinstance(callee.expr, NameExpr)
+            and f"{callee.expr.name}.{callee.name}" in TPDICT_NAMES
+        )
+
+    def process_typeddict(self, lvalue: NameExpr, rvalue: CallExpr) -> None:
+        if self._state != EMPTY:
+            self.add("\n")
+        if isinstance(rvalue.args[1], DictExpr):
+            items_list = cast(List[Tuple[StrExpr, Expression]], rvalue.args[1].items)
```

Comment:
Is this `cast` safe? What if you had something like this (which would obviously be invalid, but shouldn't make stubgen crash):

```python
T = TypeDict("T", {"a": int, **{"b": str, "c": bytes}})
```

Reply:
You are right. It is not safe. I just followed the same logic of the namedtuple implementation (which no longer uses cast in my other PR). I can check if the type is not correct and generate `T: Incomplete`. What do you think?

Reply:
> I can check if the type is not correct and generate `T: Incomplete`. What do you think?

SGTM!